lustrefs-exporter/src/routes.rs

```diff
@@ -0,0 +1,404 @@
+        lnetctl_cmd
+            .args(["net", "show", "-v", "4"])
+            .kill_on_drop(true);
+
+        let lnetctl = cmd_hdl2.output_unchecked(&mut lnetctl_cmd).await?;
+
+        let lnetctl_stats = std::str::from_utf8(&lnetctl.stdout)?;
+
+        let records = parse_lnetctl_output(lnetctl_stats)?;
+
+        Ok::<_, Error>(MetricGroup::Lustre(records))
+    });
+
+    // Collect additional LNet stats
+    let cmd_hdl2 = cmd_hdl.clone();
+    cmd_set.spawn(async move {
+        let mut lnetctl_stats_output_cmd = Command::new("lnetctl");
+        lnetctl_stats_output_cmd
+            .args(["stats", "show"])
+            .kill_on_drop(true);
+
+        let lnetctl_stats_output = cmd_hdl2
+            .output_unchecked(&mut lnetctl_stats_output_cmd)
+            .await?;
+
+        let records = parse_lnetctl_stats(std::str::from_utf8(&lnetctl_stats_output.stdout)?)?;
+
+        Ok::<_, Error>(MetricGroup::Lustre(records))
+    });
+
+    cmd_set
+}
+
+/// Main metrics scraping endpoint handler for the Prometheus exporter.
+///
+/// This function serves as the primary HTTP handler for the `/metrics` endpoint,
+/// collecting and formatting Lustre filesystem metrics in Prometheus format.
+/// It orchestrates the collection of both standard Lustre statistics and optional
+/// jobstats data based on query parameters.
+///
+/// # Arguments
+///
+/// * `Query(params)` - Query parameters extracted from the HTTP request
+/// * `State(state)` - Shared application state containing the command handler
+///
+/// # Query Parameters
+///
+/// * `jobstats` - Optional boolean parameter to enable jobstats collection
+///   (e.g., `/metrics?jobstats=true`)
+///
+/// # Returns
+///
+/// * `Ok(Response<Body>)` - HTTP response with Prometheus-formatted metrics
+/// * `Err(Error)` - Error if metric collection or formatting fails
+///
+/// # Processing Flow
+///
+/// 1. **Initialize**: Creates a new Prometheus registry and default metrics structures
+/// 2. **Conditional Jobstats**: If `jobstats=true`, collects and registers jobstats metrics
+/// 3. **Standard Metrics**: Always collects standard Lustre and LNet statistics
+/// 4. **Registration**: Registers all populated metrics with the registry
+/// 5. **Encoding**: Encodes metrics in Prometheus text format
+/// 6. **Response**: Returns HTTP 200 response with metrics as body
+///
+/// # Performance Considerations
+///
+/// - Jobstats collection can be resource-intensive and is optional but will
+///   be run within a spawned task.
+/// - Standard metrics collection runs commands concurrently for efficiency
+/// - Only metrics with actual data are registered to keep output clean
+pub async fn scrape(
+    Query(params): Query<Params>,
+    State(state): State<Arc<AppState>>,
+) -> Result<Response<Body>, Error> {
+    let mut registry = Registry::default();
+
+    // Build the lustre stats
+    let otel_jobstats = JobstatMetrics::default();
+    let mut opentelemetry_metrics = Metrics::default();
```

Comment:
I missed a few otel|opentelemetry references that Raphael snuck *everywhere*

Reply:
I'm pretty sure everything is accounted for now. All of the snapshots appear to be in good shape.